CMakeLists.txt

```diff
@@ -644,74 +557,48 @@ if(USE_GCC OR USE_CLANG OR USE_INTELCC OR USE_QCC)
       int main(void) { return 0; }" HAVE_GCC_FVISIBILITY)
   cmake_pop_check_state()
 
-  check_c_compiler_flag(-Wshadow HAVE_GCC_WSHADOW)
-  if(HAVE_GCC_WSHADOW)
-    target_compile_options(sdl-global-options INTERFACE "-Wshadow")
-  endif()
-
-  check_c_compiler_flag(-Wunused-local-typedefs HAVE_GCC_WUNUSED_LOCAL_TYPEDEFS)
-  if(HAVE_GCC_WUNUSED_LOCAL_TYPEDEFS)
-    target_compile_options(sdl-global-options INTERFACE "-Wno-unused-local-typedefs")
-  endif()
-
   if(APPLE)
-    cmake_push_check_state()
     # FIXME: don't use deprecated declarations
-    check_c_compiler_flag(-Wno-error=deprecated-declarations HAVE_WNO_ERROR_DEPRECATED_DECLARATIONS)
-    if(HAVE_WNO_ERROR_DEPRECATED_DECLARATIONS)
-      target_compile_options(sdl-build-options INTERFACE "-Wno-error=deprecated-declarations")
-    endif()
-    cmake_pop_check_state()
-
-    if(CMAKE_VERSION VERSION_LESS "3.6")
-      target_link_libraries(sdl-shared-build-options INTERFACE "-Wl,-compatibility_version,${SDL_DYLIB_COMPAT_VERSION}")
-      target_link_libraries(sdl-shared-build-options INTERFACE "-Wl,-current_version,${SDL_DYLIB_CURRENT_VERSION}")
-    endif()
-  elseif(NOT OPENBSD)
-    cmake_push_check_state()
-    set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -Wl,--no-undefined")
-    check_c_compiler_flag("" HAVE_NO_UNDEFINED)
-    cmake_pop_check_state()
-    if(HAVE_NO_UNDEFINED AND NOT (USE_CLANG AND WINDOWS))
-      target_link_libraries(sdl-shared-build-options INTERFACE "-Wl,--no-undefined")
+    check_c_compiler_flag(-Wno-error=deprecated-declarations COMPILER_SUPPORTS_WNO_ERROR_DEPRECATED_DECLARATIONS)
+    if(COMPILER_SUPPORTS_WNO_ERROR_DEPRECATED_DECLARATIONS)
```

Comment:
Are we sure openbsd behaves after this?

Reply:
ugh. wrong place.

Reply:
No, because it's not tested on ci.
If we use deprecated declarations of openbsd, then we should add an exception like there is one for macos.